### PR TITLE
Feature : Add Doctor Specialisation

### LINKED ISSUE #144: As a user, I can add specialisation for a doctor
So that I can keep track of the specified  doctor's specialisation.

### PR DESCRIPTION
**Pull request by kimshitong**
Closes #144

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddDoctorSpecialisationCommand.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import seedu.address.model.Model;
+import seedu.address.model.person.doctor.Doctor;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Adds a doctor to the database.
+ */
+public class AddDoctorSpecialisationCommand extends Command {
+
+    public static final String COMMAND_WORD = "add_spec";
```
Comment (mounilsankar) [inline src/main/java/seedu/address/logic/commands/AddDoctorSpecialisationCommand.java:25]: Change command word to add_tag_d

File: src/main/java/seedu/address/logic/commands/AddDoctorSpecialisationCommand.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.doctor.Doctor;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Adds a doctor to the database.
```
Comment (mounilsankar) [inline src/main/java/seedu/address/logic/commands/AddDoctorSpecialisationCommand.java:21]: Update this javadoc

File: src/main/java/seedu/address/logic/commands/AddDoctorSpecialisationCommand.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+
+    public static final String MESSAGE_SUCCESS = "New specialisation for doctor added: %1$s";
+    public static final String MESSAGE_DUPLICATE_TAG = "This specialisation already exists on the doctor";
+
+    private final Index index;
+    private final Tag specialisation;
+
+    /**
+     * Creates an AddDoctorTagCommand to add the specified {@code Doctor}
```
Comment (mounilsankar) [inline src/main/java/seedu/address/logic/commands/AddDoctorSpecialisationCommand.java:42]: Name of command doesnt match

File: src/main/java/seedu/address/logic/parser/AddDoctorSpecialisationCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new AddDoctorSpecialisationCommand object
+ */
+public class AddDoctorSpecialisationCommandParser implements Parser<AddDoctorSpecialisationCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddCommand object for execution.
```
Comment (mounilsankar) [inline src/main/java/seedu/address/logic/parser/AddDoctorSpecialisationCommandParser.java:20]: Change AddCommand

File: src/main/java/seedu/address/logic/commands/AddSpecialisationCommand.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.doctor.Doctor;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Adds a prescription to an existing appointment in the database.
```
Comment (yanghengtang) [inline src/main/java/seedu/address/logic/commands/AddSpecialisationCommand.java:21]: `prescription to an existing appointment` change to `specialisation for a doctor`

File: src/test/java/seedu/address/logic/commands/AddSpecialisationCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+import seedu.address.model.Database;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.doctor.Doctor;
+import seedu.address.model.tag.Tag;
+import seedu.address.testutil.DoctorBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for AddDoctorSpecialisationCommand.
```
Comment (yanghengtang) [inline src/test/java/seedu/address/logic/commands/AddSpecialisationCommandTest.java:31]: `AddDoctorSpecialisationCommand ` change to `AddSpecialisationCommand `

File: src/main/java/seedu/address/logic/commands/AddSpecialisationCommand.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        return index.equals(otherAddCommand.index)
+                && specialisation.equals(otherAddCommand.specialisation);
+    }
+    @Override
+    public int hashCode() {
+        return Objects.hash(index, specialisation);
+    }
+
+    @Override
+    public String toString() {
```
Comment (yanghengtang) [inline src/main/java/seedu/address/logic/commands/AddSpecialisationCommand.java:102]: Add test for `toString()` method

Comment (kimshitong) [reply src/main/java/seedu/address/logic/commands/AddSpecialisationCommand.java:102]: Done

File: src/test/java/seedu/address/logic/commands/AddSpecialisationCommandTest.java (hunk 6 above)
Comment (kimshitong) [reply src/test/java/seedu/address/logic/commands/AddSpecialisationCommandTest.java]: Done, thanks

File: src/main/java/seedu/address/logic/commands/AddSpecialisationCommand.java (hunk 5 above)
Comment (kimshitong) [reply src/main/java/seedu/address/logic/commands/AddSpecialisationCommand.java]: Ok noted, Thanks pointing out

File: src/main/java/seedu/address/logic/parser/AddSpecialisationCommandParser.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.AddSpecialisationCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new AddDoctorSpecialisationCommand object
```
Comment (mounilsankar) [inline src/main/java/seedu/address/logic/parser/AddSpecialisationCommandParser.java:14]: Update this javadoc

**Review by mounilsankar — APPROVED** [review]
LGTM

**Pull request merged**